model/src/payout_curve.rs

```diff
@@ -43,29 +41,63 @@ mod utils;
 /// * price: BTC-USD exchange rate used to create CFD contract
 /// * quantity: Interger number of one-dollar USD contracts contained in the
 /// CFD; expressed as a Usd amount
-/// * leverage: Leveraging used by the taker
+/// * long_leverage: leverage used by the party with the long position
+/// * short_leverage: leverage used by the party with the short position
+/// * n_payouts: number of segments into which the payout curve is discretised
+/// * fee: offset applied to the curve representing a fee paid between parties
 ///
 /// ### Returns
 ///
 /// The list of [`Payout`]s for the given price, quantity and leverage.
-pub fn calculate(
+pub fn calculate_long_taker(
```

Comment:
Is the expectation that there will be a `calculate_short_taker` as well? 
Will there be separate functions for `calculate_long_maker` and `calculate_short_maker`? 

I would actually find it easier if we would move this logic on the `Cfd`. The cfd knows position and role and can decide how to call the `calculate_payout_parameters` parameters. I find the indirection with `calculate_long_taker` in between confusing.

Reply:
Fair, I can try to move these APIs to the `model` crate.